File: src/server/internal/entity/graph.go
package entity

type Node struct {
	ID      string `json:"id"`
	Title   string `json:"title"`
	Cluster int    `json:"cluster"`
	Type    string `json:"type"`
}

type Link struct {
	Source string  `json:"source"`
	Target string  `json:"target"`
	Weight float64 `json:"weight"`
}

type GraphData struct {
	Directed   bool                   `json:"directed"`
	Multigraph bool                   `json:"multigraph"`
	Graph      map[string]interface{} `json:"graph"` // Пустой объект или дополнительные поля
	Nodes      []Node                 `json:"nodes"`
	Links      []Link                 `json:"links"`
}

type DocumentSaverReq struct {
	ID    string    `json:"id"`
	Graph GraphData `json:"graph"`
}

File: src/model/internal/entity/graph.py
from typing import Dict, List, Any

from pydantic import BaseModel


class Node(BaseModel):
    id: str
    title: str
    cluster: int
    type:str

class Link(BaseModel):
    source: str
    target: str
    weight: float

class GraphData(BaseModel):
    directed: bool
    multigraph: bool
    graph: Dict[str, Any]
    nodes: List[Node]
    links: List[Link]




class Group(BaseModel):
    keys: str
    ids: List[str]